runtime/src/gpu/amd/gpu-amd.c

```diff
@@ -396,7 +418,8 @@ void chpl_gpu_impl_mem_free(void* memAlloc) {
     // see note in chpl_gpu_mem_free
     hipPointerAttribute_t res;
     ROCM_CALL(hipPointerGetAttributes(&res, memAlloc));
-    switch_context(res.device);
+    int dev_lid = dev_pid_to_lid(res.device);
+    switch_context(dev_lid);
```

Comment:
`hipSetDevice` needs a physical id and you already have one here, but you have to roundtrip it to the logical id just to call `switch_context`. This pattern makes me think there should be 2 `switch_contexts`, one that takes a logical id and one that takes a physical id.

Reply:
I decided it wasn't worth the extra function. It's only a couple of array accesses.